DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -521,6 +518,119 @@ function Get-ChocoInstalledPackage {
     Return $res
 }
 
+<#
+.Synopsis
+   Run Chocolatey executable and throws error on failure
+.DESCRIPTION
+   Run Chocolatey executable and throws error on failure
+.EXAMPLE
+   Invoke-Chocolatey "list -lo"
+.EXAMPLE
+   Invoke-Chocolatey -arguments "list -lo"
+#>
+function Invoke-Chocolatey {
+    [CmdletBinding()]
+    Param
+    (
+        # Chocolatey arguments."
+        [Parameter(Position = 0)]
+        [string]$arguments
+    )
+
+    $result = Invoke-ChocoProcess -arguments $arguments
+
+    #exit codes reference: https://docs.chocolatey.org/en-us/choco/commands/install#exit-codes
+    switch ($result.exitcode) {
+        0 {
+            #most widely used success exit code
+            $result.output.Split("`n")
+            break
+        }
+        350 {
+            # pending reboot detected, no action has occurred
+            $result.output.Split("`n")
+            throw "Error: Chocolatey detected a pending reboot from a previous installation. You can modify your DSC and use the PendingReboot DSC resource to reboot before running this command."
+            break
+        }
+        1605 {
+            #(MSI uninstall) - the product is not found, could have already been uninstalled
+            $result.output.Split("`n")
+            break
+        }
+        1614 {
+            #(MSI uninstall) - the product is uninstalled
+            $result.output.Split("`n")
+            break
+        }
+        1641 {
+            #(MSI) - restart initiated
+            $result.output.Split("`n")
+            Request-DscReboot
+            break
+        }
+        3010 {
+            #(MSI, InnoSetup can be passed to provide this) - restart required
+            $result.output.Split("`n")
+            Request-DscReboot
+            break
+        }
+        default {
+            #when error, throw output as error, contains errormessage
+            throw "Error: Chocolatey command failed with exit code $($result.exitcode).`n$($result.output)" 
+        }
+    }
+}
+
+function Invoke-ChocoProcess {

```

Comment:
Is there any particular reason to use this method to create the Chocolatey process instead of `&` or `Start-Process`? Or even the `Invoke-Expression` that was originally being used?

Reply:
From what I remember from 13 months ago, I had some issues getting a reliable last exit code with `Invoke-Expression` so I used to the code from the very old and possibly abandoned PR #103 which address the same issue.